frontend/src/metabase/admin/performance/components/StrategyForm.tsx

```diff
@@ -0,0 +1,293 @@
+  const { setStatus } = useFormContext();
+  const [wasDirty, setWasDirty] = useState(false);
+
+  const selectedStrategyType = values.type;
+
+  useEffect(() => {
+    setIsDirty(dirty);
+  }, [dirty, setIsDirty]);
+
+  useEffect(() => {
+    // When form becomes dirty, reset form status to idle
+    setWasDirty(dirty);
+    if (dirty && !wasDirty) {
+      setStatus("idle");
+    }
+  }, [dirty, wasDirty, setIsDirty, setStatus]);
+
+  useEffect(() => {
+    if (selectedStrategyType === "duration") {
+      setFieldValue("unit", DurationUnit.Hours);
+    }
+  }, [selectedStrategyType, values, setFieldValue]);
+
+  return (
+    <Form
+      h="100%"
+      style={{
+        display: "flex",
+        flexDirection: "column",
+        justifyContent: "space-between",
+      }}
+    >
+      <Stack p="lg" spacing="xl" maw="27.5rem">
+        <StrategySelector targetId={targetId} />
+        {selectedStrategyType === "ttl" && (
+          <>
+            <Field
+              title={t`Minimum query duration`}
+              subtitle={t`Metabase will cache all saved questions with an average query execution time greater than this many milliseconds.`}
+            >
+              <PositiveNumberInput strategyType="ttl" name="min_duration_ms" />
+            </Field>
+            <Field
+              title={t`Cache time-to-live (TTL) multiplier`}
+              subtitle={
+                <div>
+                  {t`To determine how long each cached result should stick around, we take that query's average execution time and multiply that by what you input here. The result is how many seconds the cache should remain valid for.`}{" "}
+                  <Tooltip
+                    events={{
+                      hover: true,
+                      focus: true,
+                      touch: true,
+                    }}
+                    inline={true}
+                    styles={{
+                      tooltip: {
+                        whiteSpace: "normal",
+                      },
+                    }}
+                    label={t`If a query takes on average 120 seconds (2 minutes) to run, and you input 10 for your multiplier, its cache entry will persist for 1,200 seconds (20 minutes).`}
+                    maw="20rem"
+                    lh={1}
+                  >
+                    <Text tabIndex={0} display="inline" c="brand">
+                      Example
+                    </Text>
+                  </Tooltip>
+                </div>
+              }
+            >
+              <PositiveNumberInput strategyType="ttl" name="multiplier" />
+            </Field>
+          </>
+        )}
+        {selectedStrategyType === "duration" && (
+          <>
+            <Field title={t`Cache results for this many hours`}>
+              <PositiveNumberInput strategyType="duration" name="duration" />
+            </Field>
+            <input type="hidden" name="unit" />
```

Comment:
I'm not sure if this will work for you here, but if this field is in `initialValues` you should be able to work with it even without an invisible input.